### DISCUSSION TITLE
Automatically detect VS Code workspaces

### DISCUSSION THREAD

**Discussion by tobysmith568** [alefragnani/vscode-project-manager#612] [Ideas]
Hey :)

Being able to automatically detect all my git repositories is cool, but sometimes the directory that I intent to be opened in VS Code isn't the root directory of the repository. For example, in the tree below, you can see that the `app` directory is what should be opened in VS Code because that's where the `.vscode` directory is.

```
.
|- .git/
|    |- stuff
|
|- docs/
|    |- stuff.md
|
|- app/
    |- .vscode/
    |- src/
        |- code.whatever
```

Alongside your current locators, would you consider new a locator that looks for any `.vscode` directories and `*.code-workspace` files?

Ideally I can envision some form of hybrid. I'd quite like to be able to automatically detect git repositories and then also detect the `.vscode` directories and `*.code-workspace` files within them to be shown in the same view.

Perhaps, in the tree view, repositories can be clicked on to open their root directories, but any detected workspaces show as child nodes which can also be directly opened?

I'm interested in hearing your thoughts?
Thanks!

**Comment by alefragnani** [comment]
Hi @tobysmith568 ,

This possibility is already _partially_ supported. 

You just have to combine with the use of the `vscode` counterpart setting (`projectManager.vscode.baseFolders`). The auto-detect feature will look for folders that contains a `.vscode` inside. The only difference is that it will be displayed in a separate panel in the side bar.

Workspaces support, on the other hand, is not supported yet, but it is being tracked in #334.

Hope this helps
